platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -510,81 +511,106 @@ public String getFactorySnippet(@ApiParam(value = "Factory ID", required = true)
     }
 
     /**
-     * Generate project configuration.
+     * Generate factory containing workspace configuration.
+     * Only projects that have {@code SourceStorage} configured can be included.
      *
      * @param workspace
-     *         - workspace id.
+     *         - workspace id to generate factory from.
      * @param path
-     *         - project path.
+     *         - Optional project path. If set, only this project will be included into result projects set.
      * @throws org.eclipse.che.api.core.ApiException
-     *         - {@link org.eclipse.che.api.core.ConflictException} when project is not under source control.
+     *
      */
-    @ApiOperation(value = "Get project Factory parameters",
-                  notes = "This call returns a Factory.json that is used to create a Factory. " +
-                          "To be able to get project's configuration json file, it should be under Git")
+    @ApiOperation(value = "Construct Factory from workspace",
+                  notes = "This call returns a Factory.json that is used to create a Factory. ")
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "OK"),
-            @ApiResponse(code = 404, message = "Workspace or project not found"),
+            @ApiResponse(code = 403, message = "Access to workspace denied"),
+            @ApiResponse(code = 404, message = "Workspace not found"),
             @ApiResponse(code = 500, message = "Internal Server Error")})
     @GET
-    @Path("/{ws-id}/{path:.*}")
+    @Path("/workspace/{ws-id}")
```

Comment:
Hmmm /factory/workspace/WS_ID. Why do you add 'workspace' into path? Other services contains WS_ID but not contains word 'workspace' like /project/WS_ID. Seems that 'workspace' is redundant. WDYT?